### PR TITLE
Sidebar Component

### LINKED ISSUE #20: Create SideBar component
## Description
This task involves the creation of the Sidebar component, emphasizing flexibility in design. For the initial phase, focus solely on the design aspect, leaving integration considerations aside.

* We have two types of sidebar, Opened and Minimised. The developer has to create these types and focus on the flexibility to create new options and sections.

## Motivation
To provide users with easy access to different sections of the application and improve overall navigation experience, we need to create a sidebar component.

## Useful Links
[Figma Design](https://www.figma.com/design/xUlM0BIpE50NX1YItvSdGu/Escopo-projeto?node-id=295-4339&t=W3JcLYGYsyrLSRju-0)

## Acceptance Criteria
- [x] Creation of the Opened version
- [x] Creation of the Minimised version
- [x] Component flexibility for adding new sections and options

### PR DESCRIPTION
**Pull request by whyleonardo**
## Description
This PR focus in create a Sidebar component who displays most useful links on the application context. Initially, this component contains some basic nav links, but no fully properly configured because some pages weren't created yet. 

All the logic of this component is handled by zustand, saving his state on localStorage, targeting best UX principles. His states consists in `Open` and `Closed` and could be switched with a button o top of sidebar.

## Preview

https://github.com/Super-Patos-Hackaton/frontend/assets/94463485/f66766c7-f25b-4957-a19d-6e0a8eee41a4

## Todo

- [x] Handle all component logic with states
- [x] Create styles aiming the design on figma link
- [x] Apply tests in sidebar to check if all visions(`Opened` and `Closed`) displays correctly

### CODE REVIEW

File: src/components/nav-links-sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+
+import { Button, buttonVariants } from '@/components/ui/button';
```
Comment (RO-HSA) [inline src/components/nav-links-sidebar.tsx]: var buttonVariants não está sendo utilizada em lugar nenhum

**Review by roberto-juniorjp — DISMISSED** [review]
lgtm

File: src/components/ui/sidebar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+          <Button
+            data-testid='sidebar-toggle'
+            variant='ghost'
+            onClick={() => sidebar?.setIsOpen()}
+            className={cn(
+              'px-3.5',
+              'mb-1 text-brand-100 transition-transform duration-300 ease-in-out hover:text-brand-100',
+              sidebar?.isOpen === false
+                ? '-translate-x-[60px]'
```
Comment (RO-HSA) [inline src/components/ui/sidebar/index.tsx:56]: Negar com ! aqui não funcionaria?

```suggestion
              !sidebar?.isOpen
```

File: src/components/ui/sidebar/sidebar.test.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+describe('Sidebar Component', () => {
+  it('should render the Sidebar component', () => {
+    render(
+      <div>
+        <Sidebar navLinks={<button>Hello</button>} />
+      </div>
+    );
+
+    debug();
```
Comment (RO-HSA) [inline src/components/ui/sidebar/sidebar.test.tsx]: Esqueceu de tirar?

File: src/components/nav-links-sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            >
+              <Link href={href}>
+                <span className={cn(!sidebar?.isOpen ? '' : 'mr-4')}>
+                  <Icon size={20} />
+                </span>
+                <p
+                  className={cn(
+                    'max-w-[200px] truncate text-base',
+                    sidebar?.isOpen === false
+                      ? '-translate-x-96 opacity-0'
```
Comment (RO-HSA) [inline src/components/nav-links-sidebar.tsx:42]: Deixar a mesma sugestão do outro comentário...

```suggestion
                    !sidebar?.isOpen
```

File: src/components/nav-links-sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { navLinks } from '@/config/nav-links';
+
+import { cn } from '@/lib/utils';
+
+import { useSidebarToggle } from '@/hooks/use-sidebar-toggle';
+import { useStore } from '@/hooks/use-store';
+
+import { LogOut } from 'lucide-react';
+
+export const NavLinksSidebar = () => {
```
Comment (RO-HSA) [inline src/components/nav-links-sidebar.tsx:17]: Vocês precisam definir um padrão pra forma como declaram componentes... Roberto tem usado function clássica, mas acredito que poderíamos manter como arrow function

**Review by RO-HSA — CHANGES_REQUESTED** [review]
Só uns detalhes

File: src/components/nav-links-sidebar.tsx (hunk 1 above)
Comment (whyleonardo) [reply src/components/nav-links-sidebar.tsx]: that was supposed to throw an error on commit stage.

resolved on https://github.com/Super-Patos-Hackaton/frontend/commit/80ed22539ba4c3a92f2df06ca0c8c341202376a5!

File: src/components/ui/sidebar/sidebar.test.tsx (hunk 3 above)
Comment (whyleonardo) [reply src/components/ui/sidebar/sidebar.test.tsx]: resolved on https://github.com/Super-Patos-Hackaton/frontend/pull/22/commits/80ed22539ba4c3a92f2df06ca0c8c341202376a5!

File: src/components/nav-links-sidebar.tsx (hunk 4 above)
Comment (whyleonardo) [reply src/components/nav-links-sidebar.tsx:42]: resolved on https://github.com/Super-Patos-Hackaton/frontend/pull/22/commits/97a537da53af499c41603545094d3fb3c784d76f!

File: src/components/nav-links-sidebar.tsx (hunk 5 above)
Comment (whyleonardo) [reply src/components/nav-links-sidebar.tsx:17]: @roberto-juniorjp, this is a important topic that we need to discuss!

Comment (roberto-juniorjp) [reply src/components/nav-links-sidebar.tsx:17]: @whyleonardo i actually like arrow functions, so i have no problems with that. Let's go Arrow then!

**Review by RO-HSA — APPROVED** [review]
LGTM

**Review by roberto-juniorjp — APPROVED** [review]
lgtm

**Pull request merged**